src/macros.js

```diff
@@ -157,6 +172,54 @@ defineMacro("\\global", (context) => {
     }
 });
 
+// \newcommand{\macro}[args]{definition}
+// \renewcommand{\macro}[args]{definition}
+// TODO: Optional arguments: \newcommand{\macro}[args][default]{definition}
+const newcommand = (context, existsOK: boolean, nonexistsOK: boolean) => {
+    let arg = context.consumeArgs(1)[0];
+    if (arg.length !== 1) {
+        throw new ParseError(
+            "\\newcommand's first argument must be a macro name");
+    }
+    const name = arg[0].text;
+
+    const exists = context.isDefined(name);
+    if (exists && !existsOK) {
+        throw new ParseError(`Command ${name} already defined in \\newcommand`);
+    }
+    if (!exists && !nonexistsOK) {
+        throw new ParseError(
+            `Command ${name} not already defined in \\renewcommand`);
```

Comment:
This error message is a little confusing as well, maybe:
```
\\renewcommand{${name}}{...} is trying to redefine a command which doesn't exist.
```
Feel free to wordsmith these a bit more.  The main confusion for me was the "in" in "defined in \\(re)?newcommand".

Reply:
Edited both.